test/unit/tracks/text-track-settings.test.js

```diff
@@ -9,6 +9,16 @@ const tracks = [{
   label: 'test'
 }];
 
+const defaultSettings = {
```

Comment:
I think you've got the logic of this backwards - basic Text Tracks do not specify a foreground or background color, and nor does vtt.js when it parses them, so there needs to be a "I want text displayed this way" setting, so "default" doesn't make sense.

Some cues in some tracks do specify a color or other styling, in which case the setting option ought to be "leave it as specified in the track" for those cues which have a specification in the track, or "force it to the styling which I've specified, overriding what's in the cue/track".

So "---" or "default" doesn't make sense, although there ought to be an extra checkbox (or slide switch) for each setting saying "override track's own setting".